compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -5352,15 +5352,8 @@ static bool isOperationFPCompliant(TR::Node *parent,TR::Node *arithExprNode, TR:
    bool loadConstIsPowerOfTwoFloat = (mulConstChild->getDataType() == TR::Float) &&
                 isNZFloatPowerOfTwo(mulConstChild->getFloat());
 
-   bool isStrictfp =  s->comp()->getCurrentMethod()->isStrictFP() || s->comp()->getOption(TR_StrictFP);
```

Comment:
@zl-wang : this is the function I was mentioning to you earlier.  After the analysis, we'll bail out here (i.e., return `false`) if this is a strictfp method.  Given that strictfp is deprecated in JDK 17, presumably we should assume strictfp rules going forward.  Hence, the analysis in this function is moot.

The history of this code seems to be to catch an opportunity in an ancient benchmark where we can prove that an FMA instruction could be used and still obey Java's floating point rules.  I'm doubtful this catches many opportunities in real code.  Java 9+ introduces `Math.fma()` to make opportunities like this explicit to JIT compilers.

I propose we get rid of this code altogether.  Eliminating this code will also get rid of Java-specific code in OMR.

Reply:
A bit strange to be talking about Java levels here, but won't this code continue to be built into pre-Java 17 releases, which means OpenJ9 can't necessarily assume strictfp rules ? Of course, it would still be nice if the non strictfp support code didn't have to reside in OMR...

Reply:
@0xdaryl  agreed.  strictfp will be assumed always going forward and existing opt saw little opportunities in real workload.

Reply:
@mstoodle : that's true about it affecting other Java releases, but the "requirement" that I believe motivated this code is no longer relevant to any supported Java release.  I'm hoping we can agree to remove this logic entirely.

Reply:
Ok, I reacted to the remark that JDK17 changed it so we don't have to support it anymore, but the change you're proposing (which I confess I didn't look at last night before commenting <blush>) seems like it removes the code that needed to check for strictfp in this case anyway. Thanks for clarifying the intent.